C_CPP/Algorithmic Approaches/Searching/jump_search.md

```diff
@@ -0,0 +1,81 @@
+# Jump Search
+- Jump search is applied on sorted array.
+- In jump search we do not traverse through the whole array
+- Traversal goes in block fashion.
+- Blocks are formed and only that block is searched where the element is present
+ Suppose start is at starting of that block and end at last of that block then 
+  a block where arr[start]<target<arr[end] is searched and when found linear search is performed in that block.
+  For example, suppose we have an array arr[] of size n and block (to be jumped) size m. Then we search at the indexes arr[0], arr[m], arr[2m]…..arr[km] and so on.
+   Once we find the interval (arr[km] < x < arr[(k+1)m]), we perform a linear search operation from the index km to find the element x.
+Let’s consider the following array: (1,2,3,5,7,8,9,10,12). Length of the array is 9. Jump search will find the value of 10 with the following steps assuming that the block size to be jumped is 3. 
+
+- STEP 1: Jump from index 0 to index 3
+- STEP 2: Jump from index 3 to index 6
+- STEP 3: Jump from index 6 to index 9
+- STEP 4: Since 8 is the last index and the element at index 8 is greater than 10 thus linear search is performed from 6 to 8 index
+###What is the optimal block size to be skipped?   

```

Comment:
leave some space after '###' for the formatting

Reply:
Done